src/MediatR/Mediator.cs

```diff
@@ -41,14 +41,19 @@ public Task<TResponse> Send<TResponse>(IRequest<TResponse> request, Cancellation
         }
 
         public Task Publish<TNotification>(TNotification notification, CancellationToken cancellationToken = default)
-            where TNotification : INotification
+            where TNotification : INotification => Publish(notification, notification?.GetType(), cancellationToken);

```

Comment:
This could be `typeof(TNotification)`

Reply:
should be GetType(), as TNotification can be a interface/abstract type.